### PR TITLE
SWC-6640

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

**Comment by jay-hodgson** [comment]
Screen Shot 2024-01-10 at 2 51 00 PM

File: packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.ts [hunk 1]
```diff
@@ -54,13 +55,35 @@ export default function useDetectSSOCode(
   // state is used during OAuth based Synapse account creation (it's the username)
   const state = searchParams?.get('state')
 
+  const { accessToken } = useSynapseContext()
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.ts:58]: This is the tricky bit. If logged into Synapse and you're logging into ORCiD, then assume this is to bind the ORCiD to your account.  If not logged in and logging into ORCiD, then assume user is attempting to login via ORCiD to an account that is already bound to an ORCiD account

Comment (nickgros) [reply packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.ts:58]: Clever!

File: packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.ts [hunk 2]
```diff
@@ -80,13 +103,13 @@ export default function useDetectSSOCode(
...
           }
-          console.error('Error with Google account association: ', err)
+          console.error('Error with account login: ', err)
           if (onError) {
             onError(err.reason)
           }
         }
 
-        if (state) {
+        if (OAUTH2_PROVIDERS.GOOGLE == provider && state) {
```
Comment (nickgros) [inline packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.ts:112]: Got it, this is because we don't support account creation with ORCiD.

**Review by nickgros — APPROVED** [review]

File: packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.ts (hunk 2 above)
Comment (jay-hodgson) [reply packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.ts:112]: Exactly

**Pull request merged**